app/scripts/modules/core/config/settings.ts

```diff
@@ -0,0 +1,174 @@
+interface IAzureProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+  };
+}
+
+interface IAWSProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+    iamRole: string;
+    subnetType?: string;
+  };
+  defaultSecurityGroups: string[];
+  loadBalancers: {
+    inferInternalFlagFromSubnet: boolean;
+    certificateTypes?: string[];
+  };
+  useAmiBlockDeviceMappings: boolean;
+  classicLaunchLockout?: number;
+  metrics?: {
+    customNamespaces?: string[];
+  };
+}
+
+interface IGCEProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+    zone: string;
+  };
+}
+
+interface ITitusProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+    iamProfile: string;
+  };
+}
+
+interface IOpenStackProviderSettings {
+  defaults: {
+    account: string;
+    region: string;
+  };
+}
+
+interface IKubernetesProviderSettings {
+  defaults: {
+    account: string;
+    namespace: string;
+    proxy: string;
+  };
+}
+
+interface IAppengineProviderSettings {
+  defaults: {
+    account: string;
+    editLoadBalancerStageEnabled: boolean;
+  };
+}
```

Comment:
Alright, I agree with specific provider code in specific modules, so for now, instead of moving all the provider specific stuff out, I'll just remove that level of typing for now, since it really didn't matter, and we can revisit provider specific settings typings later.